ast/ddl.go

```diff
@@ -193,7 +193,32 @@ type ReferenceDef struct {
 
 // Restore implements Node interface.
 func (n *ReferenceDef) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+	if n.Table != nil {
+		ctx.WriteKeyWord("REFERENCES ")
+		if err := n.Table.Restore(ctx); err != nil {
+			return errors.Annotate(err, "An error occurred while splicing ReferenceDef")
+		}
+	}
+	ctx.WritePlain("(")
+	for i, indexColNames := range n.IndexColNames {
+		if err := indexColNames.Restore(ctx); err != nil {
```

Comment:
Missing commas between columns. Please add this test case:

```sql
    FOREIGN KEY (product_category, product_id)
      REFERENCES product(category, id)
```

Reply:
Thank you